src/databricks/labs/ucx/installer/workflows.py

```diff
@@ -463,10 +462,15 @@ def _is_nightly():
         ci_env = os.getenv("TEST_NIGHTLY")
         return ci_env is not None and ci_env.lower() == "true"
 
-    def _get_test_purge_time(self) -> str:
-        if self._is_nightly():
-            return get_test_purge_time(TEST_NIGHTLY_CI_RESOURCES_PURGE_TIMEOUT)
-        return get_test_purge_time()
+    @classmethod
+    def _get_test_purge_time(cls) -> str:
+        # Duplicate of mixins.fixtures.get_test_purge_time(); we don't want to import pytest as a transitive dependency.
+        timeout = TEST_NIGHTLY_CI_RESOURCES_PURGE_TIMEOUT if cls._is_nightly() else TEST_RESOURCE_PURGE_TIMEOUT
+        now = datetime.now(timezone.utc)
+        purge_deadline = now + timeout
+        # Round UP to the next hour boundary: that is when resources will be deleted.
+        purge_hour = purge_deadline + (datetime.min.replace(tzinfo=timezone.utc) - purge_deadline) % timedelta(hours=1)
```

Comment:
I find the following logic complicated to round up to an hour: `(datetime.min.replace(tzinfo=timezone.utc) - purge_deadline) % timedelta(hours=1)`